### PR TITLE
[WIP] Eventloop deprecation

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR deals with the deprecation of event loop from asyncio library.

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [z] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/fetchai/connections/local/connection.py [hunk 1]
```diff
@@ -104,13 +104,14 @@ def __init__(
         self._loop = loop if loop is not None else asyncio.new_event_loop()
         self._thread = Thread(target=self._run_loop, daemon=True)
 
-        self._in_queue = asyncio.Queue(loop=self._loop)  # type: asyncio.Queue
+        self._in_queue = None  # type: Optional[asyncio.Queue]
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/local/connection.py:107]: Are the logic changes necessary?

File: aea/multiplexer.py [hunk 2]
```diff
@@ -159,7 +161,6 @@ def __init__(
         self._send_loop_task = None  # type: Optional[asyncio.Task]
 
         self._loop: asyncio.AbstractEventLoop = loop if loop is not None else asyncio.new_event_loop()
-        self._lock: asyncio.Lock = asyncio.Lock(loop=self._loop)
```
Comment (DavidMinarsch) [inline aea/multiplexer.py:162]: Why the removal of the lock?

Comment (DavidMinarsch) [reply aea/multiplexer.py:162]: At a minimum init should specify the variable as None; defining new instance variables elsewhere is not recommended.

File: aea/helpers/async_utils.py [hunk 3]
```diff
@@ -407,7 +406,6 @@ def start(self) -> bool:
         self._is_running = False
         self._got_result = False
         self._set_loop()
-        self._completed_event = asyncio.Event(loop=self._loop)
```
Comment (DavidMinarsch) [inline aea/helpers/async_utils.py:410]: Why no longer neeed?

**Review by DavidMinarsch — COMMENTED** [review]
Are the new APIs compatible all the way back to 3.6? Currently we support >= 3.6 (https://github.com/valory-xyz/open-aea/blob/main/setup.py#L127) although CI just checks for >= 3.7. Also 3.6 is no longer supported (https://endoflife.date/python) so from my perspective it would be ok to only support >=3.7 going forward.

File: packages/fetchai/connections/local/connection.py (hunk 1 above)
Comment (angrybayblade) [reply packages/fetchai/connections/local/connection.py:107]: Most of the changes I did where ported from this PR: https://github.com/fetchai/agents-aea/pull/2679/files I didn't gave enough thought, If you want I can take a deeper look why these changes were made

**Pull request merged**